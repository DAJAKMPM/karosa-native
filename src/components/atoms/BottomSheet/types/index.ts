/**
 *
 * @format
 *
 */

export type PropsType = {
  children: React.ReactNode;
  height: number;
};
