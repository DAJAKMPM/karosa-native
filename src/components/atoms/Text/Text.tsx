/**
 *
 * Text
 * @format
 *
 */

import React, { FC } from "react";
import { Text as RnText } from "react-native-elements";

import type { PropsType } from "./types";
import TextStyles from "./styles";

const Text: FC<PropsType> = (props) => {
  const { text, customStyle, numberOfLines, ellipsizeMode } = props;

  return (
    <RnText
      style={[TextStyles.text, customStyle]}
      numberOfLines={numberOfLines}
      ellipsizeMode={ellipsizeMode}
    >
      {text}
    </RnText>
  );
};

export default Text;
