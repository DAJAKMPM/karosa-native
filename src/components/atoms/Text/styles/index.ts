/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

const TextStyles = StyleSheet.create({
  text: {
    ...theme.textRegular,
    color: theme.colors.dark20,
    fontWeight: "400",
  },
});

export default TextStyles;
