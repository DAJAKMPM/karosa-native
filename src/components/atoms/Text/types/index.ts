/**
 *
 * @format
 *
 */

import { StyleProp, TextStyle } from "react-native";

type EllipsizeModeType = "head" | "middle" | "tail" | "clip";

export type PropsType = {
  text: string;
  customStyle?: StyleProp<TextStyle>;
  numberOfLines?: number;
  ellipsizeMode?: EllipsizeModeType;
};
