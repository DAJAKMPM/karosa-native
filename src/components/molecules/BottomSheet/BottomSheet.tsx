/**
 *
 * BottomSheet
 * @format
 *
 */

import React, { forwardRef, Ref } from "react";
import RBSheet from "react-native-raw-bottom-sheet";

import type { PropsType } from "./types";
import BottomSheetStyles from "./styles";

const BottomSheet = forwardRef((props: PropsType, ref: Ref<RBSheet>) => {
  const { children, height } = props;

  return (
    <RBSheet
      ref={ref}
      height={height}
      closeOnDragDown={true}
      closeOnPressMask={false}
      customStyles={{ container: BottomSheetStyles.container }}>
      {children}
    </RBSheet>
  );
});

export default BottomSheet;
