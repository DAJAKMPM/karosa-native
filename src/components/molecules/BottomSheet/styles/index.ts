/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";

const BottomSheetStyles = StyleSheet.create({
  container: {
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    padding: 5,
  },
});

export default BottomSheetStyles;
