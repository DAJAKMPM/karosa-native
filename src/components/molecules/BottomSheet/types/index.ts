/**
 *
 * @format
 *
 */

import { ReactElement } from "react";

export type PropsType = {
  children: ReactElement | ReactElement[];
  height: number;
  onOpen?(): void;
  onClose?(): void;
};
