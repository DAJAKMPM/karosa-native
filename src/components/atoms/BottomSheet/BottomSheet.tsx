/**
 *
 * BottomSheet2
 * @format
 *
 */

import React from "react";
import RBSheet from "react-native-raw-bottom-sheet";

import { Props } from "./types";

const BottomSheet = React.forwardRef(
  ({ height, children }: Props, ref: React.Ref<RBSheet>) => {
    return (
      <RBSheet
        ref={ref}
        closeOnDragDown={true}
        closeOnPressMask={false}
        height={height}
        customStyles={{
          container: {
            borderTopLeftRadius: 10,
            borderTopRightRadius: 10,
            padding: 5,
          },
        }}>
        {children}
      </RBSheet>
    );
  }
);

export default BottomSheet;
