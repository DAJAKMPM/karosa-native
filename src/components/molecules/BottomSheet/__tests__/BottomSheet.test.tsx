import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Text from "@app/atoms/Text";
import BottomSheet from "../BottomSheet";

describe("BottomSheet Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <BottomSheet height={200}>
        <Text text="test" />
      </BottomSheet>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
