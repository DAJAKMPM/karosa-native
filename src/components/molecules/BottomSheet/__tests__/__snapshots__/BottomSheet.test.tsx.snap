// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomSheet Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#00000077",
              "flex": 1,
            },
            undefined,
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "opacity": 1,
          }
        }
      />
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "height": 0,
            "overflow": "hidden",
            "padding": 4,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#ccc",
                  "borderRadius": 5,
                  "height": 5,
                  "margin": 10,
                  "width": 35,
                },
                undefined,
              ]
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#1C1C1C",
              "fontFamily": "SFProText-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 18.75,
            }
          }
        >
          test
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
